fili-core/src/test/groovy/com/yahoo/bard/webservice/util/UtilsSpec.groovy

```diff
@@ -83,4 +86,100 @@ class UtilsSpec extends Specification {
         expect:
         Stream.of(1, 2, 3).reduce() { a, b -> Utils.getMinValue(a, b) }.get() == 1
     }
+
+    def "Canonicalization of object with context preserved without recursion"() {
```

Comment:
I'd be tempted to condense these into a @Unroll, where test with the where blocks having a comment field:
Similar to the https://github.com/yahoo/fili/blob/246596e04a10c0e4cde701dbd216d606e412799d/fili-core/src/test/groovy/com/yahoo/bard/webservice/async/jobs/stores/JobRowFilterSpec.groovy#L37-L73